src/test/pybind/test_rbd.py

```diff
@@ -1243,6 +1245,36 @@ def test_follower_write(self):
             for offset in [0, IMG_SIZE // 2]:
                 read = image2.read(offset, 256)
                 eq(data, read)
+    def test_acquire_release_lock(self):
+        with Image(ioctx, image_name) as image:
+            image.lock_acquire(RBD_LOCK_MODE_EXCLUSIVE)
+            image.lock_release()
+
+    def test_break_lock(self):
+        blacklist_rados = Rados(conffile='')
+        blacklist_rados.connect()
+        try:
+            blacklist_ioctx = blacklist_rados.open_ioctx(pool_name)
+            try:
+                with Image(ioctx, image_name) as image, \
+                     Image(blacklist_ioctx, image_name) as blacklist_image:
+                    blacklist_image.lock_acquire(RBD_LOCK_MODE_EXCLUSIVE)
+                    assert_raises(ReadOnlyImage, image.lock_acquire,
+                                  RBD_LOCK_MODE_EXCLUSIVE)
+
+                    lock_owners = list(image.lock_get_owners())
+                    eq(1, len(lock_owners))
+                    eq(RBD_LOCK_MODE_EXCLUSIVE, lock_owners[0]['mode'])
+                    image.lock_break(RBD_LOCK_MODE_EXCLUSIVE,
+                                     lock_owners[0]['owner'])
+
+                    image.lock_acquire(RBD_LOCK_MODE_EXCLUSIVE)
+                    assert_raises(ConnectionShutdown,
+                                  blacklist_image.lock_release())
```

Comment:
@dillaman The parenthesis after blacklist_image.lock_release look wrong. The function name should be passed, not the result of its call.

And fixing this reveals that actually ConnectionShutdown is not raised for me because ReleaseRequest::handle_unlock clears the return value:
```
2017-01-04 17:53:54.013369 7f28467fc700 10 librbd::exclusive_lock::ReleaseRequest: send_close_object_map
2017-01-04 17:53:54.013371 7f28467fc700 10 librbd::object_map::UnlockRequest: 0x7f2854000cf0 send_unlock: oid=rbd_object_map.17e02ae8944a
2017-01-04 17:53:54.013814 7f28597fa700 10 librbd::object_map::UnlockRequest: 0x7f2854000cf0 handle_unlock: r=-108
2017-01-04 17:53:54.013820 7f28597fa700 -1 librbd::object_map::UnlockRequest: failed to release object map lock: (108) Cannot send after transport endpoint shutdown
2017-01-04 17:53:54.013825 7f28597fa700 10 librbd::exclusive_lock::ReleaseRequest: handle_close_object_map: r=0
2017-01-04 17:53:54.013828 7f28597fa700 10 librbd::exclusive_lock::ReleaseRequest: send_unlock
2017-01-04 17:53:54.013829 7f28597fa700 10 librbd::ExclusiveLock: 0x7f28500187b0 handle_releasing_lock
2017-01-04 17:53:54.014153 7f28597fa700 10 librbd::exclusive_lock::ReleaseRequest: handle_unlock: r=-108
2017-01-04 17:53:54.014157 7f28597fa700 -1 librbd::exclusive_lock::ReleaseRequest: failed to unlock: (108) Cannot send after transport endpoint shutdown
2017-01-04 17:53:54.014163 7f28597fa700 10 librbd::ImageState: 0x2104a00 handle_prepare_lock_complete
2017-01-04 17:53:54.014168 7f28467fc700 10 librbd::ExclusiveLock: 0x7f28500187b0 handle_release_lock: r=0
```